storage/client/src/main/java/com/azure/storage/blob/StorageAsyncClient.java

```diff
@@ -82,10 +78,52 @@ public static StorageClientBuilder storageClientBuilder() {
      *     A {@link ContainerAsyncClient} object pointing to the specified container
      */
     public ContainerAsyncClient getContainerAsyncClient(String containerName) {
+        return new ContainerAsyncClient(new AzureBlobStorageBuilder()
+            .url(Utility.appendToURLPath(getAccountUrl(), containerName).toString())
+            .pipeline(storageAsyncRawClient.azureBlobStorage.httpPipeline()));
+    }
+
+    /**
+     * Creates a new container within a storage account. If a container with the same name already exists, the operation
+     * fails. For more information, see the
+     * <a href="https://docs.microsoft.com/rest/api/storageservices/create-container">Azure Docs</a>.
+     *
+     * @param containerName Name of the container to create
+     * @return A response containing a {@link ContainerAsyncClient} used to interact with the container created.
+     */
+    public Mono<Response<ContainerAsyncClient>> createContainer(String containerName) {
```

Comment:
If we're going to be able to create containers from the parent client, we need to discuss how to provide a similar experience for blobs. Since we're prioritizing block blob work right now, that's a bit tricky, as block blobs are the only blob type that don't need to be explicitly created upfront, but can just be uploaded to. Additionally, put blob rest calls will update (overwrite) an existing blob, while create container will fail if the container already exists, so that may influence our design.

Reply:
As James pointed out, `createBlob()` method on a container client doesn't make sense for a block blob. Not sure what to do, leaving it for discussions after I'm OOF.